staging/src/k8s.io/pod-security-admission/policy/check_allowPrivilegeEscalation.go

```diff
@@ -63,21 +64,74 @@ func CheckAllowPrivilegeEscalation() Check {
 
 func allowPrivilegeEscalation_1_8(podMetadata *metav1.ObjectMeta, podSpec *corev1.PodSpec) CheckResult {
 	var badContainers []string
+	var badInitContainers []string
+	var badEphemeralContainers []string
 	visitContainers(podSpec, func(container *corev1.Container) {
 		if container.SecurityContext == nil || container.SecurityContext.AllowPrivilegeEscalation == nil || *container.SecurityContext.AllowPrivilegeEscalation {
 			badContainers = append(badContainers, container.Name)
 		}
 	})
 
-	if len(badContainers) > 0 {
+	visitInitContainers(podSpec, func(container *corev1.Container) {
+		if container.SecurityContext == nil || container.SecurityContext.AllowPrivilegeEscalation == nil || *container.SecurityContext.AllowPrivilegeEscalation {
+			badInitContainers = append(badInitContainers, container.Name)
+		}
+	})
+
+	visitEphemeralContainers(podSpec, func(container *corev1.Container) {
+		if container.SecurityContext == nil || container.SecurityContext.AllowPrivilegeEscalation == nil || *container.SecurityContext.AllowPrivilegeEscalation {
+			badEphemeralContainers = append(badEphemeralContainers, container.Name)
+		}
+	})
+
+	if len(badContainers) > 0 || len(badInitContainers) > 0 || len(badEphemeralContainers) > 0 {
+		var errList field.ErrorList
+		for _, badContainer := range badContainers {
+			restrictedFieldErr := RestrictedFieldErr{
+				Path:          "spec.containers[*].securityContext.allowPrivilegeEscalation",
+				ContainerName: badContainer,
+				ForbiddenValues: []string{
+					"true",
+				},
+			}
+			errList = append(errList, field.Invalid(field.NewPath(restrictedFieldErr.Path), restrictedFieldErr, ""))
```

Comment:
I think a `field.Forbidden` error would be more appropriate here, and in most places. In a couple instances, `field.Required` might fit better.